tutorials/python/python-type-driven-development/tutorial.md

```diff
@@ -0,0 +1,213 @@
+---
+title: Type annotations in Python
+published: 2019-08-01
+tags: Python, typing, types, tdd
+language: python
+author-name: Carlos Villavicencio
+twitter-profile: po5i
+github-profile: po5i
+description: Learn how to improve the software quality and readability in our Python code by using the built-in typing system as it complements other software development processes like type-driven development and test-driven development.
+---
+
+_Type-Driven Development_ is a technique in which we plan our program development by adding types first and then write the code that satisfies that type. It brings many benefits such as increasing the robustness, accuracy, testability, readability, and extensibility of your code by taking advantage of the language’s type system.
+
+In a [previous article][types-vs-tests], we discussed the benefits of working with _Type-Driven Development_ and the importance of static type checking at compilation time for strongly typed languages such as Java or Haskell. Type-checking strengthens our code, reduces the number of tests we need to write which decreases our dependency on them no matter if the paradigm is object-oriented or functional.
```

Comment:
One could argue that this is not correct as it is possible and easy to just trick Java's type system

> strongly typed languages such as Java

Reply:
Good call. Do you think removing Java will be enough? or maybe replace it with another lang?

Reply:
What about a footnote acknowledging that Java might not be strongly typed as there are ways to skip the type system. But, for the sake of the tutorial, it is fine to consider it like that

Reply:
I think that can be confusing to the reader, mostly because we're not addressing Java or covering a type system topic. But an interesting topic that can be expanded or elaborated after.